lighthouse-core/gather/gatherers/dobetterweb/domnodes.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Gathers all DOM nodes on the page.
+ */
+
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+class AllDOMNodes extends Gatherer {
+  /**
+   * @param {!Object} options
+   * @return {!Promise<!Array<!Object>>}
+   */
+  afterPass(options) {
+    return options.driver.querySelectorAll('body, body /deep/ *');
```

Comment:
Gmail is 4000-5000 elements (depending on the view, sometimes it's 15k on long email threads). This github review page is ~1500. My facebook timeline is 2300 when loaded. I got my numbers sampling a bunch of popular websites.

btw the /deep/ trick won't work with Shadow DOM v1 because I don't think deep will match in there. You'll need to write your own traversal that crawls the entire document. It's unfortunate we don't have any tooling for querySelectorAll with Shadow DOM v1 yet. @hayatoito